src/conn/conn_chunkcache.c

```diff
@@ -0,0 +1,344 @@
+    cursor->set_key(cursor, entry->name, entry->id, entry->file_offset);
+    cursor->set_value(cursor, entry->cache_offset, entry->data_sz);
+
+    return (cursor->insert(cursor));
+}
+
+/*
+ * __chunkcache_metadata_delete --
+ *     Remove an entry from the chunk cache metadata file.
+ */
+static int
+__chunkcache_metadata_delete(WT_CURSOR *cursor, WT_CHUNKCACHE_METADATA_WORK_UNIT *entry)
+{
+    cursor->set_key(cursor, entry->name, entry->id, entry->file_offset);
+
+    return (cursor->remove(cursor));
+}
+
+/*
+ * __chunkcache_metadata_pop_work --
+ *     Pop a work unit from the queue. The caller is responsible for freeing the returned work unit
+ *     structure.
+ */
+static void
+__chunkcache_metadata_pop_work(WT_SESSION_IMPL *session, WT_CHUNKCACHE_METADATA_WORK_UNIT **entryp)
+{
+    WT_CONNECTION_IMPL *conn;
+
+    conn = S2C(session);
+
+    __wt_spin_lock(session, &conn->chunkcache_metadata_lock);
+    if ((*entryp = TAILQ_FIRST(&conn->chunkcache_metadataqh)) != NULL) {
+        TAILQ_REMOVE(&conn->chunkcache_metadataqh, *entryp, q);
+        --conn->chunkcache_queue_len;
+    }
+    __wt_spin_unlock(session, &conn->chunkcache_metadata_lock);
+
+    WT_STAT_CONN_INCR(session, chunkcache_metadata_work_units_dequeued);
+}
+
+/*
+ * __chunkcache_metadata_work --
+ *     Pop chunk cache work items off the queue, and write out the metadata.
+ */
+static int
+__chunkcache_metadata_work(WT_SESSION_IMPL *session)
+{
+    WT_CHUNKCACHE_METADATA_WORK_UNIT *entry;
+    WT_CURSOR *cursor;
+    WT_DECL_RET;
+
+    entry = NULL;
+
+    WT_ERR(__wt_open_cursor(session, WT_CC_METAFILE_URI, NULL, NULL, &cursor));
+
+    for (int i = 0; i < WT_CHUNKCACHE_METADATA_MAX_WORK; i++) {
+        if (!__chunkcache_metadata_run_chk(session))
+            break;
+
+        __chunkcache_metadata_pop_work(session, &entry);
+        if (entry == NULL)
+            break;
+
+        if (entry->type == WT_CHUNKCACHE_METADATA_WORK_INS)
+            WT_ERR(__chunkcache_metadata_insert(cursor, entry));
+        else if (entry->type == WT_CHUNKCACHE_METADATA_WORK_DEL)
+            WT_ERR_NOTFOUND_OK(__chunkcache_metadata_delete(cursor, entry), false);
+        else {
+            __wt_verbose_error(
+              session, WT_VERB_CHUNKCACHE, "got invalid event type %d\n", entry->type);
+            ret = -1;
+            goto err;
+        }
+
+        __wt_free(session, entry);
+        entry = NULL;
+    }
+
+err:
+    if (cursor != NULL)
+        WT_TRET(cursor->close(cursor));
+    if (entry != NULL)
+        __wt_free(session, entry);
+    return (ret);
+}
+
+/*
+ * __chunkcache_metadata_server --
+ *     Dispatch chunks of work (or stop the server) whenever we're signalled to do so.
+ */
+static WT_THREAD_RET
+__chunkcache_metadata_server(void *arg)
+{
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+    uint64_t cond_time_us;
+    bool signalled;
+
+    session = arg;
+    conn = S2C(session);
+    cond_time_us = 1 * WT_MILLION;
```

Comment:
why do we need the 1 * here?

Reply:
It's not strictly necessary - I just found that it reads a bit more naturally. I can drop it if you dislike it.

Reply:
I haven't found any other place we do this in the code, I would vote for just WT_MILLION - to me it's clear enough.

Reply:
Yep, fair enough. Removed in 5b54ca5.